### PR TITLE
Implementing methods and helper functions

### LINKED ISSUE #9: Alex Chow - Working on methods and helper functions
* [x]  changeAssignment Grade
* [ ] change Assignment Max Grade
* [x] getAssignmentIndex
* [x] getWeightFromType
* [x] getOnlyCategoryTotal

### PR DESCRIPTION
**Pull request by VermilionAlex**
Added new vector for assignment names and getAssignmentIndex. changeAssignmentGrade should be fine, but not tested yet.
Need getAssignmentIndex now to make to use other functions/methods easier.

### CODE REVIEW

**Review by klattuada — COMMENTED** [review]
Index Method looks good

**Review by klattuada — APPROVED** [review]
Index Method looks good

File: gradebook.cpp [hunk 1]
```diff
@@ -12,11 +12,26 @@ double getCourseGrade(int option);
...
       
 void changeAssignmentName(std::string p_assignment_name, string p_new_name);
 void changeAssignmentType(std::string p_assignment_name, string p_new_type);
-void changeAssignmentGrade(std::string p_assignment_name, int p_new_score);
+void changeAssignmentGrade(std::string p_assignment_name, int p_new_score){
+    int index = getAssignmentIndex(p_assignment_name);
+    this-> score_obtained.at(index) = p_new_score;
+
+}
 void changeAssignmentMaxGrade(std::string p_assignment_name, int p_new_max_score);
```
Comment (pmajdalani) [inline gradebook.cpp:35]: all methods should have "grade_Book::" in between the method type and method name, this should be added before the merge is accepted.

**Review by pmajdalani — DISMISSED** [review]

File: gradebook.cpp (hunk 1 above)
Comment (pmajdalani) [reply gradebook.cpp:35]: JD just let me know that the changes have been fixed, on his side so i am going to resolve this.

**Pull request merged**